frontend/src/tests/applicant-tests.js

```diff
@@ -126,7 +128,92 @@ export function applicantTests(api) {
         expect(resp.payload).toContainObject(newApplicant);
     });
 
-    it.todo("delete an applicant with no associated assignments");
+    it("delete an applicant with no associated assignments", async () => {
+        // upsert and delete an applicant without a session
+        const newApplicant1 = {
+            first_name: "Tommy3",
+            last_name: "Smith3",
+            utorid: "smith3",
+        };
+        const resp = await apiPOST(`/admin/applicants`, newApplicant1);
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(applicantPropTypes, resp.payload);
+        Object.assign(newApplicant1, resp.payload);
+
+        const resp2 = await apiPOST(`/admin/applicants/delete`, {
+            id: newApplicant1.id,
+        });
+        expect(resp2).toHaveStatus("success");
+
+        const resp3 = await apiGET(`/admin/applicants`);
+        expect(resp3).toHaveStatus("success");
+        expect(resp3.payload.map((x) => x.id)).not.toContain(newApplicant1.id);
+
+        // upsert and delete an applicant with a session
+        const newApplicant2 = {
+            first_name: "Tommy4",
+            last_name: "Smith4",
+            utorid: "smith4",
+        };
+        const resp4 = await apiPOST(
+            `/admin/sessions/${session.id}/applicants`,
+            newApplicant2
+        );
+        expect(resp4).toHaveStatus("success");
+        checkPropTypes(applicantPropTypes, resp4.payload);
+        Object.assign(newApplicant2, resp4.payload);
+
+        const resp5 = await apiPOST(`/admin/applicants/delete`, {
+            id: newApplicant2.id,
+        });
+        expect(resp5).toHaveStatus("success");
+
+        const resp6 = await apiGET(`/admin/sessions/${session.id}/applicants`);
+        expect(resp6).toHaveStatus("success");
+        expect(resp6.payload.map((x) => x.id)).not.toContain(newApplicant2.id);
+    });
+
+    it("fail to delete an applicant with an associated assignment", async () => {
+        const newApplicant1 = {
+            first_name: "Tommy4",
+            last_name: "Smith4",
+            utorid: "smith4",
+        };
+        const resp = await apiPOST(`/admin/applicants`, newApplicant1);
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(applicantPropTypes, resp.payload);
+        Object.assign(newApplicant1, resp.payload);
```

Comment:
I may be mistaken, but I believe that you can have multiple assignments to a position for each applicant. So, we could use the database seeder instead of retesting the applicant insert.

Reply:
Look at the seed data. There should be an applicant with an associated assignment already. That should make this tests one or two lines long in total.